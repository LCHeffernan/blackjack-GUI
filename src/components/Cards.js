import React from "react";
import PropTypes from "prop-types";
import Card from "./Card";
import "../styles/cards.css";

const Cards = ({ cards }) => {
  return (
    <div className="player-hand">
      {cards.map((card) => (
        <Card key={`${card.cardRank}${card.cardSuit}`} card={card} />
      ))}
    </div>
  );
};

Cards.propTypes = {
  cards: PropTypes.arrayOf(
    PropTypes.shape({
      cardRank: PropTypes.string,
      cardSuit: PropTypes.string,
      cardValue: PropTypes.number,
    })
  ).isRequired,
};

export default Cards;
