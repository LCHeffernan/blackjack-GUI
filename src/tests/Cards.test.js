import React, { render } from "@testing-library/react";
import Cards from "../components/Cards";

describe("Cards", () => {
  const validProps = {
    cards: [
      {
        cardRank: "A",
        cardSuit: "Mock",
        cardValue: 11,
      },
      {
        cardRank: "K",
        cardSuit: "Mock",
        cardValue: 10,
      },
    ],
  };
  it("renders correctly", () => {
    const { asFragment } = render(<Cards cards={validProps.cards} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
