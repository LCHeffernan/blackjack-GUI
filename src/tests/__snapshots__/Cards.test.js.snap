// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards renders correctly 1`] = `
<DocumentFragment>
  <div
    class="player-hand"
  >
    <div
      class="card-container"
    >
      <div
        class="top-left-rank"
      >
        A
      </div>
      <div
        class="suit"
      >
        Mock
      </div>
      <div
        class="bottom-right-rank"
      >
        A
      </div>
    </div>
    <div
      class="card-container"
    >
      <div
        class="top-left-rank"
      >
        K
      </div>
      <div
        class="suit"
      >
        Mock
      </div>
      <div
        class="bottom-right-rank"
      >
        K
      </div>
    </div>
  </div>
</DocumentFragment>
`;
